Assets/Scripts/ConnectionManagement/ConnectionState/HostingState.cs

```diff
@@ -25,15 +25,17 @@ class HostingState : ConnectionState
 
         public override void Enter()
         {
-            var gameState = UnityEngine.Object.Instantiate(m_ConnectionManager.GameState);
-
-            gameState.Spawn();
+            foreach (var prefab in m_ConnectionManager.NetworkObjectsToSpawnWhenServerStarted)
```

Comment:
ConnectionManagement shouldn't be in charge of spawning objects like this. This should be done by the appropriate game state, no?
cc @pdeschain who's working in that area as well.

Reply:
I agree, however changing this is not in the scope of this PR in my opinion